### PR TITLE
SOC-5615 : [Cluster] Cannot start second node after finish social mig…

### PR DESCRIPTION
**Pull request by aboughzela**
On actual social migration process, remove social workspace
The second node try to synchronise index from first one ( recovry mode) for all workspace (include social ws)
Fail to start repository container.

### CODE REVIEW

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/RDBMSMigrationManager.java [hunk 1]
```diff
@@ -297,7 +297,7 @@ public void run() {
                 timeToCleanupSpaces = System.currentTimeMillis() - t;
               }
 
-              if (MigrationContext.isIdentityCleanupDone()&& MigrationContext.isSpaceCleanupDone() || forceRemoveJCR) {
+              if (!clusterMode && (MigrationContext.isIdentityCleanupDone()&& MigrationContext.isSpaceCleanupDone() || forceRemoveJCR)) {
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/RDBMSMigrationManager.java]: Does it mean that in cluster mode the social will never be removed ?

Comment (aboughzela) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/RDBMSMigrationManager.java]: * The social workspace is configured on the other node , remove the workspace from first node broken start other node.
* restart the node ==> workspace is re-created (https://jira.exoplatform.org/browse/SOC-5537)
* social data is well cleaned by MigrationCleanup
The clean way to remove this workspace : remove social  configuration on the next release GA

Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/RDBMSMigrationManager.java]: And what is the clean way with 4.4 ?
You mean we should not delete the social workspace in 4.4, and do it in the next version ?

**Comment by aboughzela** [comment]
Update PR : Remove workspace social should be done next version

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateActivitiesTest.java [hunk 2]
```diff
@@ -141,7 +141,8 @@ public void testMigrationActivities() throws Exception {
     assertTrue(getOrCreateSettingValue(MigrationContext.SOC_RDBMS_IDENTITY_MIGRATION_KEY));
     assertTrue(getOrCreateSettingValue(MigrationContext.SOC_RDBMS_IDENTITY_CLEANUP_KEY));
 
-    verifyWorkspaceRemoved();
+    /** Remove workspace should be done next version
+     verifyWorkspaceRemoved();**/
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateActivitiesTest.java]: To be removed. We just instead open a Jira issue targeted to 5.x.
Also the method verifyWorkspaceRemoved() should be removed.

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateEmptyDataTest.java [hunk 3]
```diff
@@ -42,6 +42,7 @@ public void testMigrationEmptyData() throws Exception {
     assertTrue(getOrCreateSettingValue(MigrationContext.SOC_RDBMS_IDENTITY_MIGRATION_KEY));
     assertTrue(getOrCreateSettingValue(MigrationContext.SOC_RDBMS_IDENTITY_CLEANUP_KEY));
 
-    verifyWorkspaceRemoved();
+    /** Remove workspace should be done next version
+    verifyWorkspaceRemoved();**/
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateEmptyDataTest.java]: To be removed.

**Comment by aboughzela** [comment]
Done remove unused method

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/RDBMSMigrationManager.java [hunk 4]
```diff
@@ -297,32 +292,8 @@ public void run() {
...
-                    MigrationContext.setDone(true);
-                  } else {
-                    LOG.warn("Social workspace is not removeable, so it is not removed.");
-                  }
-
-                } catch (RepositoryException ex) {
-                  LOG.error("Can not remove social workspace", ex);
-                }
+              if (MigrationContext.isIdentityCleanupDone()&& MigrationContext.isSpaceCleanupDone() || forceRemoveJCR){
+                updateSettingValue(MigrationContext.SOC_RDBMS_MIGRATION_STATUS_KEY, Boolean.TRUE);
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/RDBMSMigrationManager.java]: Don't you forget MigrationContext.setDone(true); ?

Comment (aboughzela) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/updater/RDBMSMigrationManager.java]: Of course

**Pull request closed without merging**